packages/grid/x-data-grid/src/components/columnHeaders/GridBaseColumnHeaders.tsx

```diff
@@ -23,8 +23,9 @@ const GridColumnHeadersRoot = styled('div', {
   slot: 'ColumnHeaders',
   overridesResolver: (props, styles) => styles.columnHeaders,
 })<{ ownerState: OwnerState }>({
-  position: 'relative',
-  overflow: 'hidden',
+  position: 'sticky',
```

Comment:
This is conflicting with row pinning, which also uses `position: sticky` 🤔
https://deploy-preview-10059--material-ui-x.netlify.app/x/react-data-grid/row-pinning/

https://github.com/mui/mui-x/assets/13808724/6cfa8a94-66f0-44ec-8e9d-13cfae81dcf5